controllers/object_controls.go

```diff
@@ -3748,17 +3820,63 @@ func getDaemonsetControllerRevisionHash(ctx context.Context, daemonset *appsv1.D
 	return hash, nil
 }
 
+// TransformDRADriverController transforms nvidia-dra-driver-controller deployment with required config as per ClusterPolicy
+func TransformDRADriverController(obj *appsv1.Deployment, spec *gpuv1.ClusterPolicySpec) error {
+	config := spec.DRADriver
+	image, err := gpuv1.ImagePath(&config)
+	if err != nil {
+		return err
+	}
+	obj.Spec.Template.Spec.Containers[0].Image = image
```

Comment:
Two comments

- Let's extract the image index by name, instead of a hardcoded reference to the zero-index
- move `obj.Spec.Template.Spec.Containers[i]` to a variable to avoid redundancy. I know we've done this in the past, but we should break away from this practice

Reply:
Updated to address your feedback.